torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1197,53 +1197,58 @@ def sample_inputs_linalg_vector_norm(op_info, device, dtype, requires_grad, **kw
+    """Operator information for 'universal binary functions (binary ufuncs).'
+    These are functions of two tensors with common properties like:
+      - they are elementwise functions
+      - the output shape is determined by the input shape
+      - they typically have method and inplace variants
+      - they typically support the out kwarg
+      - they typically have NumPy or SciPy references
+    See NumPy's universal function documentation
+    (https://numpy.org/doc/stable/reference/ufuncs.html) for more details
+    about the concept of ufuncs.
+    """
+    pass
+
+
+def sample_inputs_binary_pwise(
+        op_info, device, dtype, requires_grad, op_kwargs=None, lhs_make_tensor_kwargs=None, **rhs_make_tensor_kwargs
+):
+    if op_kwargs is None:
+        op_kwargs = {}
+    if lhs_make_tensor_kwargs is None:
+        lhs_make_tensor_kwargs = {}
+
+    shapes = [
+        ((1,), ()),
+        ((S,), ()),
+        ((1,), (S,)),
+        ((S,), (S,)),
+        ((S, 1), (S,)),
+        ((1, S), (S,)),
+        ((M, S), (S,)),
+        ((M, S), (M, S)),
+        ((1, M, S), (S,)),
+        ((1, M, S), (M, S)),
+        ((M, 1, S), (M, S)),
+        ((1, M, S), (1, M, S)),
+        ((S, M, S), ()),
+        ((S, M, S), (S, M, S)),
+        ((M, 1, S), (1, M, S)),
     ]
-    tests_with_lhs_broadcasting = [
-        ((S, S), (S, S, S), True),
-        ((), (S, S, S), True),
-        ((S, 1, S), (M, S), True),
+
+    return [
+        SampleInput(
+            make_tensor(shape_a, device=device, dtype=dtype, requires_grad=requires_grad, **lhs_make_tensor_kwargs),
+            args=(make_tensor(shape_b, device=device, dtype=dtype, **rhs_make_tensor_kwargs),),
+            kwargs=op_kwargs,
+        )
+        for shape_a, shape_b in shapes
     ]
-    test_cases = tests_list + tests_with_lhs_broadcasting  # type: ignore[operator]
-    samples = []
-    for first_shape, shape_or_scalar, broadcasts_input in test_cases:
-        arg = shape_or_scalar
-
-        if isinstance(shape_or_scalar, tuple):
-            exclude_zero = kwargs.get('rhs_exclude_zero', False)
-            arg = make_tensor(shape_or_scalar, device=device, dtype=dtype,
-                              requires_grad=requires_grad, exclude_zero=exclude_zero)
-        samples.append(SampleInput(make_tensor(first_shape, device=device, dtype=dtype,
-                                               requires_grad=requires_grad),
-                                   args=(arg,), kwargs=extra_kwargs,
-                                   broadcasts_input=broadcasts_input))
-    # Adds an extra sample using "alpha" if it's passed in kwargs
-    if 'alpha' in kwargs:
-        a = make_tensor((S, S, S), device=device, dtype=dtype, requires_grad=requires_grad)
-        b = make_tensor((S, S, S), device=device, dtype=dtype, requires_grad=requires_grad)
-        extra_kwargs['alpha'] = kwargs['alpha']
-        sample = SampleInput(a, args=(b,), kwargs=extra_kwargs)
-        samples.append(sample)
```

Comment:
I've refrained to add any special handling for the `alpha` kwarg, because only very few operators have it and this is a generic function.